yt_idefix/data_structures.py

```diff
@@ -353,13 +367,19 @@ def _parse_inifile(self) -> None:
             if any(_ != "u" for _ in vals[3::3]):
                 msg_elems.append(f"found non-uniform block(s) in direction {ax}")
 
-    @abstractmethod
     def _parse_definitions_header(self) -> None:
+        self.parameters["definitions"] = {}
```

Comment:
I see you defined `self.parameters["definitions"]` both here and in `PlutoVtkDataset`. Is it plausible to define it once in their super class `IdefixDataset`, before calling `_parse_definitions_header`

Reply:
Well it's a tradeoff. I think it's probably better to repeat this one line rather than coupling the classes even more and make the flow ever so slightly harder to follow (admittedly it's currently a mess).

Reply:
Ok!